sdk/storage/Azure.Storage.Blobs/README.md

```diff
@@ -33,77 +44,131 @@ Blob storage is designed for:
-
-var download = await blob.DownloadAsync();
-using (var file = File.Create("BlockDestination.txt"))
-{   
-    await download.Value.Content.CopyToAsync(file);
+// Get a reference to the public blob at https://aka.ms/bloburl
+BlobClient blob = new BlobClient(new Uri("https://aka.ms/bloburl"));
+
+// Download the blob
+BlobDownloadInfo download = blob.Download();
+using (FileStream file = File.OpenWrite("hello.jpg"))
+{
+    download.Content.CopyTo(file);
 }
 ```
 
 ### Enumerating blobs
+
 ```c#
-string connectionString = <connection_string>;
-var service = new BlobServiceClient(connectionString);
-var container = service.GetBlobContainerClient("mycontainer");
+string connectionString = "<connection_string>";
+
+// Get a reference to a container named "sample-container"
+BlobContainerClient container = new BlobContainerClient(connectionString, "sample-container");
 
-string marker;
-do
+// List all of its blobs
+foreach (BlobItem blob in container.GetBlobs())
 {
-    var response = await container.ListBlobsFlatSegmentAsync(marker);
-    foreach (var blob in response.Value.BlobItems)
-    {
-        Console.WriteLine(blob.Name);
-    }
-    marker = response.Value.NextMarker;
+    Console.WriteLine(blob.Name);
 }
-while (!string.IsNullOrEmpty(marker));
 ```
 
+### Authenticating with Azure.Identity
+
+The [Azure Identity library](identity) provides easy Azure Active Directory support for authentication.
+
+```c#
+using Azure.Identity;
+using Azure.Security.KeyVault.Secrets;
+
+// Create a BlobServiceClient that will authenticate through Active Directory
+Uri accountUri = new Uri("https://MYSTORAGEACCOUNT.blob.core.windows.net/");
+BlobServiceClient client = new BlobServiceClient(accountUri, new DefaultAzureCredential());
+```
+
+Learn more about enabling Azure Active Directory for authentication with Azure Storage in [our documentation](storage_ad) and [our samples](#next-steps).
+
 ## Troubleshooting
+
 All Blob service operations will throw a
 [StorageRequestFailedException][StorageRequestFailedException] on failure with
-helpful [`ErrorCode`s][error_codes].
+helpful [`ErrorCode`s][error_codes].  Many of these errors are recoverable.
+
+```c#
+string connectionString = "<connection_string>";
+
+// Try to create a container named "sample-container" and avoid any potential race
+// conditions that might arise by checking if the container exists before creating
+BlobContainerClient container = new BlobContainerClient(connectionString, "sample-container");
+try
+{
+    container.Create();
+}
+catch (StorageRequestFailedException ex)
+    when (ex.ErrorCode == BlobErrorCode.ContainerAlreadyExists)
+{
+    // Ignore any errors if the container already exists
+}
+```
 
 ## Next steps
-Get started with our [Blob samples][samples].
+
+Get started with our [Blob samples][samples]:
+
+1. [Hello World](samples/Sample01a_HelloWorld.cs): Upload, download, and list blobs (or [asynchronously](samples/Sample01b_HelloWorldAsync.cs))

```

Comment:
Hmm it might be too late to tell people that we also have async APIs in here... Consider adding an explanation in the Examples section where you say what type of examples they can find and that the package also includes async APIs. Something like: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/keyvault/Azure.Security.KeyVault.Keys#examples

Reply:
That's a good idea.  Fixed.